CiscoESA/cisco-esa/ingest/parser.yml

```
@@ -0,0 +1,313 @@
+      - set:
+          '@timestamp': "{{parse_end_date.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time == null and parse_end_date != null}}"
+
+  set_cef_header_fields:
+  # CEF:Version|Device Vendor|Device Product|Device Version|Device Event Class ID|Name|Severity|[Extension]
+    actions:
+
+      - set:
+          observer.vendor: '{{parsed_event.message.DeviceVendor}}'
+          observer.type: '{{parsed_event.message.DeviceProduct}}'
+          observer.version: '{{parsed_event.message.DeviceVersion}}'
+          rule.id: '{{parsed_event.message.DeviceEventClassID}}'
+
+      - set:
+          # Match only for agentSeverity of type int
+          event.severity: '{{parsed_event.message.Severity}}'
+        filter: "{{parsed_event.message.Severity | re_match('[1-9]|10')}}"
+
+
+  set_cef_extension_fields:
+    actions:
+
+      - set:
+          event.action: '{{parsed_event.message.act}}'
+          network.protocol: '{{parsed_event.message.app}}'
+          event.reason: '{{parsed_event.message.msg}}'
+          file.name: '{{parsed_event.message.fname}}'
+          file.size: '{{parsed_event.message.fsize}}'
+          source.ip: '{{parsed_event.message.src or parsed_event.message.sourceAddress}}'
+          source.user.name: '{{parsed_event.message.suser}}'
+          destination.domain: "{{parsed_event.message.get('destination DnsDomain')}}"
+          service.name: "{{parsed_event.message.get('destination ServiceName')}}"
+          destination.nat.ip: "{{parsed_event.message.get('destination Translated Address')}}"
+          destination.nat.port: "{{parsed_event.message.get('destination TranslatedPort')}}"
+
+      - translate:
+        dictionary:
+          0: 'inbound'
+          1: 'outbound'
+        mapping:
+          parsed_event.message.deviceDirection: network.direction
+
+      - set:
+          host.name: "{{parsed_event.message.dvchost or {parsed_event.message.get('deviceDns Domain')}}"
+          host.id: "{{parsed_event.message.get('device ExternalId')}}"
+          log.syslog.facility.name: "{{parsed_event.message.deviceFacility}}"
+          observer.ingress.interface.name: "{{parsed_event.message.get('deviceInbound Interface')}}"
+          host.domain: "{{parsed_event.message.get('deviceNt Domain')}}"
+          observer.egress.interface.name: "{{parsed_event.message.get('Device Outbound Interface')}}"
+          process.name: "{{parsed_event.message.get('deviceProcess Name')}}"
+          server.nat.ip: "{{parsed_event.message.get('device Translated Address')}}"
+          destination.domain: "{{parsed_event.message.dhost or parsed_event.message.dntdom}}"
+          destination.port: "{{parsed_event.message.dpt}}"
+          destination.ip: "{{parsed_event.message.dst}}"
+          event.timezone: "{{parsed_event.message.dtz}}"
+          destination.user.id: "{{parsed_event.message.duid}}"
+          destination.user.name: "{{parsed_event.message.duser}}"
```

It seems that [`destination.user.email`](https://www.elastic.co/guide/en/ecs/8.6/ecs-user.html#field-user-email) is more suitable
```suggestion
          destination.user.email: "{{parsed_event.message.duser}}"
```

I think we can use a grok to extract the `destination.user.name` and the `destination.user.domain` from the email